agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/home/HomeViewModel.kt

```diff
@@ -25,11 +32,12 @@ class HomeViewModel(
 
     init {
         viewModelScope.launch {
-            state = state.copy(
-                profileInitials = formatProfileNameUseCase.invoke(
-                    sessionStorage.get()?.fullName.orEmpty()
-                )
-            )
+            setProfileInitials()
+            snapshotFlow { state.selectedDate }
+                .distinctUntilChanged()
+                .collect {
```

Comment:
With states, I'd always use collectLatest to be sure you're dealing with the most up-to-date emissions and not with one that doesn't reflect the current UI state anymore.